### ISSUE TITLE
Smokeping document improvement

### ISSUE THREAD

**Issue by eikik87** [librenms/librenms#5376]
This is only adding information that might be helpful. At least I had a issue with it for some time. Laf on irc helped me getting this working correctly.

I was always editing the ./gen_smokeping.php script, so it would generate the 'librenms.conf' file with the required text so smokeping website would compile the files correctly. Otherwise the website just runs into errors.

*** Targets ***

probe = FPingNormal

remark = VFIS Smokeping

"

There is better way to do this, and that is to create new text file under "/etc/smokeping/config.d/". I called it 'extra'.

I then input the text above into that file. Then in the "/etc/smokeping/config" I would include that file above the include line for librenms.conf file. So it would look something like this.

@include /etc/smokeping/config.d/General
@include /etc/smokeping/config.d/Alerts
@include /etc/smokeping/config.d/Database
@include /etc/smokeping/config.d/Presentation
@include /etc/smokeping/config.d/Probes
@include /etc/smokeping/config.d/Slaves
@include /etc/smokeping/config.d/extra
@include /etc/smokeping/config.d/librenms.conf

This will make smokeping parse the missing text and the website compiles correctly. So now I don't have to edit the script and my auto updates can work again.

**Comment by laf** [comment]
You can update documentation yourself by clicking edit on the page you were following and then follow the instructions on github.

**Comment by geordish** [comment]
The default smokeping config has that configuration in a file called "Targets". This is included in /etc/smokeping/config by default (at least on Debian...)
```
cat /etc/smokeping/config.d/Targets
*** Targets ***

probe = FPing
```

```
cat /etc/smokeping/config
@include /etc/smokeping/config.d/General
@include /etc/smokeping/config.d/Alerts
@include /etc/smokeping/config.d/Database
@include /etc/smokeping/config.d/Presentation
@include /etc/smokeping/config.d/Probes
@include /etc/smokeping/config.d/Slaves
@include /etc/smokeping/config.d/Targets
```

The instructions as followed for me result in just adding the extra config at the bottom.

Did you accidentally remove the Targets config when installing?

**Comment by laf** [comment]
Thanks @geordish

**Comment by eikik87** [comment]
Yes that is correct. I removed it in /etc/smokeping/config file most likely when I was figuring this problem out why the website didn't get compiled correctly. And when I did manually edit on the gen_smokeping.php script I just wanted librenms.conf file, not the targets. The Targets file already had some information in it, not just the default text. So I guess me not understanding how smokeping parses the include lines in order caused this issue. So maybe just a information about how smokeping parses might make people like me not fucking this up.
